packages/core/src/MachineNode.ts

```diff
@@ -238,7 +249,8 @@ export class MachineNode<
    */
   public get initialState(): State<TContext, TEvent, TStateSchema, TTypestate> {
     this._init();
-    return getInitialState(this);
+    const nextState = getInitialState(this);
+    return macrostep(nextState, null as any, this);
```

Comment:
`null as any` is our representation of an "initial event" that should throw an error if improperly accessed